frontend/src/components/userprofile/userprofile.jsx

```diff
@@ -0,0 +1,31 @@
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import './UserProfile.css';
+
+const UserProfile = ({ userId }) => {
+  const [user, setUser] = useState(null);
+
+  useEffect(() => {
+    axios.get(`/api/users/${userId}`)

```

Comment:
This needs the full api url. 

You could also put this in a .env file so that we could do it like how you have it. Otherwise it's trying to get that url. 

And I don't even think we have that route you're trying to access? Might want to double check.